pkg/webhooks/networkpolicies/networkpolicies.go

```diff
@@ -0,0 +1,204 @@
+	managedNamespaces              string = `(^openshift-.*|kube-system)`
+)
+
+var (
+	timeout                          int32 = 2
+	allowedUsers                           = []string{"system:admin", "backplane-cluster-admin"}
+	sreAdminGroups                         = []string{"system:serviceaccounts:openshift-backplane-srep"}
+	privilegedServiceAccountGroupsRe       = regexp.MustCompile(privilegedServiceAccountGroups)
+	scope                                  = admissionregv1.NamespacedScope
+	rules                                  = []admissionregv1.RuleWithOperations{
+		{
+			Operations: []admissionregv1.OperationType{
+				admissionregv1.Create,
+				admissionregv1.Update,
+				admissionregv1.Delete,
+			},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{"networking.k8s.io"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"networkpolicies"},
+				Scope:       &scope,
+			},
+		},
+	}
+	log = logf.Log.WithName(WebhookName)
+)
+
+// networkpoliciesruleWebhook validates a networkpolicy change
+type networkpoliciesruleWebhook struct {
+	s runtime.Scheme
+}
+
+// We just need a runtime object to get the namespace
+type networkPolicy struct {
+	runtime.Object
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+}
+
+// NewWebhook creates the new webhook
+func NewWebhook() *networkpoliciesruleWebhook {
+	scheme := runtime.NewScheme()
+	return &networkpoliciesruleWebhook{
+		s: *scheme,
+	}
+}
+
+// Authorized implements Webhook interface
+func (s *networkpoliciesruleWebhook) Authorized(request admissionctl.Request) admissionctl.Response {
+	//Implement authorized next
+	return s.authorized(request)
+}
+
+func (s *networkpoliciesruleWebhook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+
+	np, err := s.renderNetworkPolicy(request)
+
+	if err != nil {
+		log.Error(err, "Could not render a NetworkPolicy from the incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+
+	if hookconfig.IsPrivilegedNamespace(np.GetNamespace()) {
+		log.Info(fmt.Sprintf("%s operation detected on managed namespace: %s", request.Operation, np.GetNamespace()))
+		if isAllowedUser(request) {
+			ret = admissionctl.Allowed(fmt.Sprintf("User can do operations on NetworkPolicies"))
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
+		for _, group := range request.UserInfo.Groups {
+			if privilegedServiceAccountGroupsRe.Match([]byte(group)) {
+				ret = admissionctl.Allowed("Privileged service accounts do operations on NetworkPolicies")
+				ret.UID = request.AdmissionRequest.UID
+				return ret
+			}
+		}
+
+		ret = admissionctl.Denied(fmt.Sprintf("Prevented from accessing Red Mat managed resources. This is in an effort to prevent harmful actions that may cause unintended consequences or affect the stability of the cluster. If you have any questions about this, please reach out to Red Hat support at https://access.redhat.com/support"))
```

Comment:
We can remove the `fmt.Sprintf()` call since we aren't formatting anything:
```suggestion
		ret = admissionctl.Denied("Prevented from accessing Red Mat managed resources. This is in an effort to prevent harmful actions that may cause unintended consequences or affect the stability of the cluster. If you have any questions about this, please reach out to Red Hat support at https://access.redhat.com/support")
```